### ISSUE TITLE
Vulnerability with react-native v0.72.6 - Updating to v0.73 generate npm peer dependency error

### ISSUE THREAD

**Issue by quickseries** [nrwl/nx#20644]
### Current Behavior

Snyk reported the following vulnerability with `react-native v0.72.6`, suggesting an update to `v0.73`.

<img width="888" alt="Screenshot 2023-12-07 at 10 59 13 AM" src="https://github.com/nrwl/nx/assets/28966611/bcf7c808-cee0-4b7f-a363-db67abc6bf37">

Unfortunately, it would seems that `@nx/react-native@17.1.3` depends on `react-native@">= 0.72.0 < 0.73.0"` which results in an error when attempting to install this package update.

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @spacemates/source@0.0.0
npm ERR! Found: react-native@0.73.0
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.73.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@">= 0.72.0 < 0.73.0" from @nx/react-native@17.1.3
npm ERR! node_modules/@nx/react-native
npm ERR!   dev @nx/react-native@"17.1.3" from the root project
```

### Expected Behavior

Update the peer dependencies to allow this vulnerability to be fixed.

### Steps to Reproduce

1. In your `package.json` file, change `react-native v0.72.6` to `react-native v0.73.0`
2. Delete `node_modules` directory
3. Delete `package_lock.json` file
4. Run `npm i`

### Nx Report

```shell
Node   : 18.17.0
   OS     : darwin-arm64
   npm    : 9.8.1
   
   nx (global)        : 17.1.3
   nx                 : 17.1.3
   @nx/js             : 17.1.3
   @nx/jest           : 17.1.3
   @nx/linter         : 17.1.3
   @nx/eslint         : 17.1.3
   @nx/workspace      : 17.1.3
   @nx/detox          : 17.1.3
   @nx/devkit         : 17.1.3
   @nx/eslint-plugin  : 17.1.3
   @nx/react          : 17.1.3
   @nx/react-native   : 17.1.3
   @nrwl/tao          : 17.1.3
   @nx/web            : 17.1.3
   typescript         : 5.2.2
```

### Failure Logs

```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @spacemates/source@0.0.0
npm ERR! Found: react-native@0.73.0
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.73.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@">= 0.72.0 < 0.73.0" from @nx/react-native@17.1.3
npm ERR! node_modules/@nx/react-native
npm ERR!   dev @nx/react-native@"17.1.3" from the root project
```

### Package Manager Version

npm 9.8.1

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by 1abhishekpandey** [comment]
We are waiting for this to be fixed by nx. Refer to this issue: https://github.com/nrwl/nx/issues/20644.

**Comment by leosvelperez** [comment]
This was fixed by https://github.com/nrwl/nx/pull/20896 and released in Nx 18.0.0.
